test/plugins/cxxrandom.lua

```diff
@@ -0,0 +1,65 @@
+local rng = require('plugins.cxxrandom')
+
+function test.cxxrandom_distributions()
+    rng.normal_distribution:new(0,5)
+    rng.real_distribution:new(-1,1)
+    rng.int_distribution:new(-20,20)
+    rng.bool_distribution:new(0.00000000001)
+    rng.num_sequence:new(-1000,1000)
+    -- no errors, no problem
+end
+
+function test.cxxrandom_seed()
+    local nd = rng.normal_distribution:new(0,500000)
+    local e1 = rng.MakeNewEngine(1)
+    local e2 = rng.MakeNewEngine(1)
+    local e3 = rng.MakeNewEngine(2)
+    local g1 = rng.crng:new(e1, true, nd)
+    local g2 = rng.crng:new(e2, true, nd)
+    local g3 = rng.crng:new(e3, true, nd)
+    expect.eq(g1:next(), g2:next())
+    expect.ne(g1:next(), g3:next())
+end
+
+function test.cxxrandom_ranges()
+    local e1 = rng.MakeNewEngine(1)
+    local g1 = rng.crng:new(e1, true, rng.normal_distribution:new(0,1))
+    local g2 = rng.crng:new(e1, true, rng.real_distribution:new(-5,5))
+    local g3 = rng.crng:new(e1, true, rng.int_distribution:new(-5,5))
+    local g4 = rng.crng:new(e1, true, rng.num_sequence:new(-5,5))
+    for i = 1, 10000 do
+        local a = g1:next()
+        local b = g2:next()
+        local c = g3:next()
+        local d = g4:next()
+        expect.ge(a, -5)
```

Comment:
Is there a deterministic limit to how high/low the normal distribution can get?

Reply:
I'd have to dig into the c11 standard to find out what they say about such things. But from a mathmatical standpoint, it'd be pretty wild to be five sigma out: https://cdn-images-1.medium.com/max/2600/1*IdGgdrY_n_9_YfkaCh-dag.png

But also, all the tests are seeded.